File: 2016/gara2/ricarica/att/ricarica.c
#include <stdio.h>
#include <assert.h>

#define MAXN 100000

int ricarica(int N, int M, int A[], int B[]) {
    // Mettete qui il codice della soluzione
    return 42;
}


int A[MAXN], B[MAXN];

int main() {
    FILE *fr, *fw;
    int N, M, i;

    fr = fopen("input.txt", "r");
    fw = fopen("output.txt", "w");
    assert(2 == fscanf(fr, "%d %d", &N, &M));
    for(i=0; i<N; i++)
        assert(2 == fscanf(fr, "%d %d", &A[i], &B[i]));

    fprintf(fw, "%d\n", ricarica(N, M, A, B));
    fclose(fr);
    fclose(fw);
    return 0;
}

File: ed2/gara2/ricarica/att/ricarica.pas
function ricarica(N, M: longint; var A, B: array of longint): longint;
begin
    (* Mettete qui il codice della soluzione *)
    ricarica := 42;
end;


const
    MAXN = 100000;
var
    N, M, i : longint;
    A, B    : array[0..MAXN-1] of longint;
    fr, fw  : text;

begin
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
    reset(fr);
    rewrite(fw);
    readln(fr, N, M);
    for i:=0 to N-1 do
        readln(fr, A[i], B[i]);

    writeln(fw, ricarica(N, M, A, B));
    close(fr);
    close(fw);
end.